app/utils.py

```diff
@@ -13,3 +16,22 @@ async def wrap_http_error_in_502(base_msg: str = "Error in FinOps for Cloud"):
             status_code=status.HTTP_502_BAD_GATEWAY,
             detail=f"{base_msg}: {e.response.status_code} - {e.response.text}.",
         ) from e
+
+
+@contextlib.asynccontextmanager
+async def wrap_exc_in_http_response(
+    exc_cls: type[Exception],
+    error_msg: str | None = None,
+    status_code: int = status.HTTP_400_BAD_REQUEST,
+):
+    try:
+        yield
```

Comment:
this context managers should be sync since we do not perform asyncio neither in `__aenter__` nor in `__aexit__`